### DISCUSSION TITLE
opendos-64... really?

### DISCUSSION THREAD

**Discussion by stsp** [dosemu2/dosemu2#2271] [General]
https://osm.fandom.com/wiki/64-DOS
I think its a fake though.

**Comment by stuaxo** [comment]
A funny mock up.

Strange ports seem to happen on leaked code though - like when they ported the DOS support in Windows to 64 bit.

As for 64 bit DOS, I'd bet on FDPP making it to hardware or UEFI before anyone ports MSDOS to 64 bit, but who knows.

**Comment by jschwartzenberg** [comment]
It will be interesting to see how large language models will deal with this. Currently Copilot points to FDPP when asking about 64-DOS. When asking for opendos-64 it points to https://github.com/SoilFoil64/OpenDOS

**Comment by stsp** [comment]
There is a very little use in porting
fdpp to uefi, as then you also need
v86 (native or under vtx), bios, vga
and all that.
Porting dosbox to uefi is more useful,
but then its unclear how to format
dos partitions. Or limit it to a subdir
of an uefi partition only? Maybe that
can work if uefi has a rich set of usb
and network drivers, on top of which
dosbox can emulate input and net.
Dosemu2 might be difficult to port
to uefi as it lacks softmmu mode
(while dosbox iirc lacks hardmmu
mode)

Пятница, 20 сентября 2024 г получено от Stuart Axon:

**Reply by jschwartzenberg** [reply]
Sure strange, but still a bigger chance that it would happen with FDPP than with MS-DOS ;)

In general if one needs modern hardware with DOS, I would suggest buying something with Coreboot and loading SeaBIOS. There are multiple places selling Coreboot laptops nowadays, for instance: https://novacustom.com/

**Reply by stsp** [reply]
> Sure strange, but still a bigger chance that it would happen with FDPP than with MS-DOS ;)

I would bet on the opposite actually. :)
Or, more likely, it would be freedos.
This is because when you have v86
and all dos-supporting hardware, then
you can already _boot_ freedos or ms-dos.
You don't have to port it.
You'll end up with the binary blobs - a
problem for dosemu2 (hence fdpp), but
not a problem for uefi.
fdpp and comcom64 are the great estimators
of how many efforts one needs to invest
to turn dosemu1 into something dosbox-alike,
and that's all about them. :)

> In general if one needs modern hardware with DOS

I am struggling to think up why would anyone
want that. With the fully emulated environment
you can use DOS in a very feature-rich manner:
you use any network card (it is visible as ne2k
in DOS, or as a packet driver), you use your USB,
video - everything. But with bare-metal you'll
struggle finding an USB and network drivers,
not to speak about the rest. I think its rather
useless.

**Reply by Torinde** [reply]
Related: https://github.com/dosemu2/dosemu2/discussions/1800 - example use-case:

> access to obscure hardware (e.g. PCIe/USB-to-ISA bridges, PCIe-to-PCI bridges, SATA/IDE, USB, ...)


USB drivers: https://github.com/crazii/USBDDOS, https://bretjohnson.us/

**Comment by stsp** [comment]
Пятница, 20 сентября 2024 г получено от Julius Schwartzenberg:
> It will be interesting to see how large language models will deal with this. Currently Copilot points to FDPP when asking about 64-DOS.

Google points to that 64-dos joke
when asked about "64bit dos". It'd
better point to fdpp, but since fdpp
isn't bootable on its own (needs
some kernel), there is little use in
ever pointing to it.
Maybe indeed those dosemu2-related
projects (fdpp, comcom64, dj64dev)
should out-grow their dosemu2
sandbox? Otherwise I have that strange
feeling that lots of code was written
with nothing at all to demonstrate. :)

> When asking for opendos-64 it points to https://github.com/SoilFoil64/OpenDOS

What a strange project.
